sleuth/sleuth/enrich_tls.py

```diff
@@ -169,76 +107,87 @@ def tls_seclevel(scs, client_key_length, certs):
-
-        elif kex == "ECDHE" or kex == "ECDH" or kex == "ECDH_anon":
-            if client_key_length < 224:
-                kex_seclevel = 1 # avoid
-            elif client_key_length < 256:
-                kex_seclevel = 2 # legacy
-            elif client_key_length < 512:
-                kex_seclevel = 3 # acceptable
-            else:
-                kex_seclevel = 4 # recommended
-        else:
-            kex_seclevel = kex_policy[kex]
+        policy_rules = policy.rules
+        min_seclevel = min(policy.classifications.values())
+        max_seclevel = max(policy.classifications.values())
 
+        # Analyze key exchange seclevel based on key length seclevel information
+        # provided in the policy json
+        #
+        kex = params['kex']
+        kex_policy = policy_rules["kex"]
+
+        kex_seclevel = UNKNOWN
+        if kex in kex_policy:
+            if client_key_length and "client_key_length" in kex_policy[kex]:
+                kex_seclevel = min_seclevel
+                for each in kex_policy[kex]["client_key_length"]:
+                    if client_key_length > int(each):
+                        kex_seclevel = kex_policy[kex]["client_key_length"][each]
+                        break
+            elif "seclevel" in kex_policy[kex]:
+                kex_seclevel = kex_policy[kex]["seclevel"]
+
+
+        # Analyze seclevel of certificates based on algorithm and key size seclevel
+        # information given in the policy json
+        #
         if certs:
-            certs_seclevel = 4
+            certs_seclevel = max_seclevel
+            sig_policy = policy_rules["sig_alg"]
+
             for x in certs:
                 sig_alg = x['signature_algorithm']
                 sig_key_size = x['signature_key_size']
-                if sig_alg == "sha1WithRSAEncryption":
-                    if sig_key_size < 1024:
-                        tmp_seclevel = 1
-                    else:
-                        tmp_seclevel = 2
-                elif sig_alg == "sha256WithRSAEncryption":
-                    if sig_key_size < 1024:
-                        tmp_seclevel = 1
-                    elif sig_key_size < 2048:
-                        tmp_seclevel = 2
-                    else:
-                        tmp_seclevel = 3
-                elif sig_alg == "ecdsa-with-SHA256":
-                    if sig_key_size < 832:
-                        tmp_seclevel = 1
-                    else:
-                        tmp_seclevel = 3
-                elif sig_alg == "sha384WithRSAEncryption" or sig_alg == "sha512WithRSAEncryption":
-                    if sig_key_size < 1024:
-                        tmp_seclevel = 1
-                    elif sig_key_size < 2048:
-                        tmp_seclevel = 2
-                    elif sig_key_size < 3072:
-                        tmp_seclevel = 3
-                    else:
-                        tmp_seclevel = 4
+                tmp_seclevel = certs_seclevel
```

Comment:
this line is not needed because you are checking if the variable exists on line 155.